tests/test_freshness_date_parser.py

```diff
@@ -96,6 +96,10 @@ def test_relative_past_dates_with_time_as_period(self, date_string, ago, period)
         param('eight months ago', ago={'months': 8}, period='month'),
         param('six days ago', ago={'days': 6}, period='day'),
         param('five years ago', ago={'years': 5}, period='year'),
+        # Fractional units
+        param('2.5 hours', ago={'hours': 2.5}, period='day'),
+        param('10.75 minutes', ago={'minutes': 10.75}, period='day'),
+        param('1.5 days', ago={'days': 1.5}, period='day'),
```

Comment:
5. Add test cases for fractional units (tests added to 4 places within this file including with commas)